libs/header/p2p/exchange.go

```diff
@@ -270,8 +276,16 @@ func (ex *Exchange[H]) request(
 		if err != nil {
 			return nil, err
 		}
+		err = validateChainID(ex.Params.chainID, header.(H).ChainID())
+		if err != nil {
+			return nil, err
+		}
```

Comment:
As chain ID is optional, the validation should also be optional. Do validate only if chainId is not ""